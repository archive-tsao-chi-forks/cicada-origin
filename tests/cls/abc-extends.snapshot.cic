@def the_nat_type
@the (Trivial) -> Type
(_) => Nat

@class ABC [ a: Trivial, b: the_nat_type(a), c: String ]

@class ABCEFG @extends ABC [ e: Trivial, f: the_nat_type(e), g: String ]

@def abcefg @the ABCEFG
{ a: sole, b: 2, c: "c",
  e: sole, f: 5, g: "g" }

@show abcefg

@show abcefg.a
@show abcefg.b
@show abcefg.c
@show abcefg.e
@show abcefg.f
@show abcefg.g
