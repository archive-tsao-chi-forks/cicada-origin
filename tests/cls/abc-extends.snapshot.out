@the [
  a: Trivial
  b: Nat
  c: String
  e: Trivial
  f: Nat
  g: String
] {
  a: sole
  b: 2
  c: "c"
  e: sole
  f: 5
  g: "g"
}
@the Trivial sole
@the Nat 2
@the String "c"
@the Trivial sole
@the Nat 5
@the String "g"

